pygen/pygen_src/isa/riscv_instr.py

```diff
@@ -46,12 +48,12 @@ def __init__(self):
         self.imm_len = 0
 
         self.csr = None
-        self.rs2 = None
-        self.rs1 = None
-        self.rd = None
-        self.imm = BitArray(hex="0x00000000")
+        self.rs2 = vsc.rand_enum_t(riscv_reg_t)
+        self.rs1 = vsc.rand_enum_t(riscv_reg_t)
+        self.rd = vsc.rand_enum_t(riscv_reg_t)
+        self.imm = vsc.rand_uint32_t()
 
-        self.imm_mask = BitArray(hex="0xffffffff")
+        self.imm_mask = 0xffffffff
```

Comment:
why is this changed to no longer be a BitArray?

Reply:
We changed self.imm_mask from BitArray to `0xffffffff` because **self.imm** is of **vsc.rand_uint32_t()** type.

We are doing **or** operation in extend_imm function as 
```python3
    self.imm = self.imm_mask | self.imm
```
We were facing some compatibility issue with vsc.rand_uint_t() and BitArray and also on debugging we found that self.imm_mask, the width was greater than 32 bit after shift operation.
That's why we use **self.shift_t** and to maintain the imm_mask width and changed it from BitArray to normal hex value.

Reply:
Thanks @aneels3 , it makes sense. Just a quick suggestion: to be more consistent (and probably get rid of some extra methods), can't we also define self.imm_mask as vsc.uint32_t (just non-random form of vsc.rand_uint32_t) or even vsc.bit_t(width)? Probably in this case we won't face the compatibility issue anymore...

Reply:
Hi @hodjat91 
Thanks for the suggestion. We tried defining the self.imm_mask as non-random but we are getting the same results. 
I think there is an issue with shift operations in python, it actually padding the zeros after the shift operation. I am not getting why the width is increasing even after defining the width using vsc.uint32_t or vsc.bit_t(32).

Reply:
This is a good point, I was able to recreate a scenario where the `vsc.uint8_t` could be set to a 9-bit value, etc..., maybe we could open an issue to inquire further about this?

Reply:
Sure @udinator 
I will open the issue for this.